oauth2_http/java/com/google/auth/oauth2/ServiceAccountJwtAccessCredentials.java

```diff
@@ -119,6 +131,7 @@ public ServiceAccountJwtAccessCredentials(String clientId, String clientEmail,
     this.privateKey = Preconditions.checkNotNull(privateKey);
     this.privateKeyId = privateKeyId;
     this.defaultAudience = defaultAudience;
+    this.tokenCache = createCache();
```

Comment:
Shouldn't this be in more than one place?  This is a deprecated method -- something appears wrong.

Reply:
Ah - method is called - it's just weird, but appears to be Ok.

Reply:
need to validate that fromStream() somehow set's tokenCache

Reply:
I'm not sure I understand what mean.  All initialization passes through this constructor. For example:
fromStream(InputStream) -> fromStream(InputStream, URI) -> fromJson(Map, URI) -> fromPkcs8(...) -> this constructor

I'm guessing `@Deprecated` annotation is intended for external callers to use the  builder instead. 

The only entry point that doesn't end up calling this constructor is when java deserialization. That ends up calling `readObject()`, which needs to recreate the cache., which is why I factored out createCache()

Reply:
@igorbernstein2 is correct, this is just marked `@Deprecated` to point people to the builder.

Reply:
SGTM